spec/services/api/v1/npq/application_status_query_spec.rb

```diff
@@ -3,36 +3,40 @@
 require "rails_helper"
 
 RSpec.describe Api::V1::NPQ::ApplicationStatusQuery do
-  let(:participant_declaration) { create(:npq_participant_declaration) }
-  let(:npq_application) { create(:npq_application, participant_identity_id: participant_declaration.participant_profile.participant_identity.id) }
-  let(:participant_outcome) { create(:participant_outcome, participant_declaration:) }
+  let(:npq_application) { create(:npq_application, :accepted) }
+  let(:participant_profile) { npq_application.profile }
+  let(:participant_declaration) { create(:npq_participant_declaration, cpd_lead_provider: npq_application.npq_lead_provider.cpd_lead_provider, participant_profile:) }
+  let!(:participant_outcome) { create(:participant_outcome, participant_declaration:) }
   let(:service) { described_class.new(npq_application) }
+
   describe "#call" do
-    context "when a participant declaration exists" do
-      before do
-        allow(NPQApplication).to receive(:participant_declaration_finder).and_return(participant_declaration)
+    context "when a non completed participant declaration exists" do
+      it "returns nil" do
+        expect(service.call).to be_nil
       end
+    end
+
+    context "when a completed participant declaration exists" do
+      before { participant_declaration.update!(declaration_type: "completed") }
+
       context "when a participant outcome exists" do
-        before do
-          allow(ParticipantOutcome::NPQ).to receive(:latest_per_declaration).and_return(participant_outcome)
-        end
         it "returns the latest state of the participant outcome which is only one created" do
-          expect(service.call).to eq(participant_outcome.state)
+          expect(service.call).to eq("passed")
         end
       end
+
       context "when no participant outcome exists" do
-        before do
-          allow(ParticipantOutcome::NPQ).to receive(:latest_per_declaration).and_return(nil)
-        end
+        before { participant_outcome.update!(participant_declaration: create(:npq_participant_declaration)) }
+
         it "returns nil" do
           expect(service.call).to be_nil
         end
       end
     end
+
     context "when no participant declaration exists" do
-      before do
-        allow(NPQApplication).to receive(:participant_declaration_finder).and_return(nil)
-      end
+      before { npq_application.destroy }
```

Comment:
I think here we should set the declaration to `nil` rather destroy the whole application